msal/src/test/java/com/microsoft/identity/client/e2e/tests/mocked/MultipleAccountPublicClientTests.java

```diff
@@ -0,0 +1,539 @@
+        assertEquals("account.getId()", expectedUserAccountData.localAccountId, account.getId());
+        assertEquals("account.getAuthority()", expectedUserAccountData.cloud, account.getAuthority());
+        assertEquals("iss claim", expectedUserAccountData.cloud, account.getClaims().get("iss"));
+        assertEquals("tid claim", expectedUserAccountData.tenantId, account.getClaims().get("tid"));
+        assertEquals("oid claim", expectedUserAccountData.localAccountId, account.getClaims().get("oid"));
+    }
+
+    @Test
+    public void testRemoveAccountRemovesAllCredentialsFromMultipleCloudsWithSameHomeAccountId() {
+        // arrange
+        // add mock record in cache which has different home account id than the account to be removed
+        ICacheRecord cacheRecord = null;
+        final TokenResponse tokenResponse = MockTokenResponse.getMockSuccessTokenResponse();
+
+        try {
+            cacheRecord = RoboTestCacheHelper.saveTokens(tokenResponse, mMultipleAccountPCA);
+        } catch (ClientException e) {
+            fail("Unable to save tokens to cache: " + e.getMessage());
+        }
+
+        final ResultFuture<Boolean> future = new ResultFuture<>();
+
+        // act
+        try {
+            mMultipleAccountPCA.removeAccount(
+                getAccount(),
+                new IMultipleAccountPublicClientApplication.RemoveAccountCallback() {
+                    @Override
+                    public void onRemoved() {
+                        future.setResult(true);
+                    }
+
+                    @Override
+                    public void onError(@NonNull MsalException exception) {
+                        future.setException(exception);
+                    }
+                });
+        } catch (Exception e) {
+            fail(e.getMessage());
+        }
+
+        RoboTestUtils.flushScheduler();
+
+        // assert
+        SharedPreferences sharedPreferences = getSharedPreferences(SHARED_PREFERENCES_NAME);
+        final Map<String, ?> cacheValues = sharedPreferences.getAll();
+        for (String key: cacheValues.keySet()) {
+            assertFalse("Cache record found for homeAccountId of removed account",
+                    key.contains(testCaseData.homeAccountId));
+            assertTrue("Cache record found for other homeAccountId",
+                    key.contains(cacheRecord.getAccount().getHomeAccountId()));
+        }
+        assertEquals("Cache Values count", 4, cacheValues.size());
+    }
+
+    @Test
+    public void testAcquireTokenReturnsAccessTokenForCrossCloudAccount() {
+        // arrange
+        final ResultFuture<IAuthenticationResult> future = new ResultFuture<>();
+        final UserAccountData lastSignedInAccount =
+                testCaseData.userAccountsData.get(testCaseData.userAccountsData.size() -1);
+        // act
+        try {
+            mMultipleAccountPCA.acquireTokenSilentAsync
+                    (getScopes(),
```

Comment:
This initial parenthesis seems to be at a string place and not follow the existing coding style patterns